cmds/rmmod/rmmod.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Remove a module from the Linux kernel
+//
+// Synopsis:
+//	rmmod name
+//
+// Description:
+//	rmmod is a clone of rmmod(8)
+//
+// Author:
+//     Roland Kammerer <dev.rck@gmail.com>
+package main
+
+import (
+	"log"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatalf("rmmod: ERROR: missing module name.\n")
+	}
+
+	modname := os.Args[1]
+	flags := syscall.O_NONBLOCK
+	modnameptr, err := syscall.BytePtrFromString(modname)
+	if err != nil {
+		log.Fatalf("rmmod: %v\n", err)
+	}
+	ret, _, err := syscall.Syscall(syscall.SYS_DELETE_MODULE, uintptr(unsafe.Pointer(modnameptr)), uintptr(unsafe.Pointer(&flags)), 0)
```

Comment:
Instead of passing the flags as a pointer, you can just cast them to uintptr. In fact, constants in go don't have a type. So you don't even need to cast, you can just

```go
syscall.Syscall(..., syscall.O_NONBLOCK, 0)
```

Reply:
Yes, I will keep the "flags" variable, but then it will be uinitprtr(flags). Will push that soon.